### ISSUE TITLE
shared-styles.js are overwriten by lumo styles if using Routing with layout 

### ISSUE THREAD

**Issue by Skrikjo** [vaadin/flow#6408]
When route definition have layout like this: `@Route(value = "dashboard", layout = MainView.class)` then import of shared-styles is on top of imports in `generared-flow-imports.js`. That conduce, variables overwritten in shared styles have no effect, as variables loaded from lumo overwrites them.

### Here is project to reproduce issue:
[orangefont 2.zip](https://github.com/vaadin/platform/files/3579373/orangefont.2.zip)
  
### some code from prev mentioned project:
Layout class
```
@JsModule("./styles/shared-styles.js")
@PWA(name = "orange font", shortName = "orange font")
@Theme(value = Lumo.class, variant = Lumo.LIGHT)
public class MainView extends AppLayout {
    private final Tabs menu;

    public MainView() {
        menu = createMenuTabs();
        addToNavbar(menu);
    }

    private static Tabs createMenuTabs() {
        final Tabs tabs = new Tabs();
        tabs.setOrientation(Tabs.Orientation.HORIZONTAL);
        tabs.add(new Tab("Dashboard"));
        return tabs;
    }
}
```

View Class:
```
@Route(value = "dashboard", layout = MainView.class)
@RouteAlias(value = "", layout = MainView.class)
@PageTitle("Dashboard")
public class DashboardView extends VerticalLayout {
    public DashboardView() {
        add(new H3("Hello!"));
    }
}
```

shared-styles.js
```
import "@vaadin/vaadin-charts/theme/vaadin-chart-default-theme";

const $_documentContainer = document.createElement('template');

$_documentContainer.innerHTML = `
<custom-style>
  <style>
html {
      --lumo-shade-5pct: rgba(33, 33, 33, 0.05);
      --lumo-shade-10pct: rgba(33, 33, 33, 0.1);
      --lumo-shade-20pct: rgba(33, 33, 33, 0.2);
      --lumo-shade-30pct: rgba(33, 33, 33, 0.3);
      --lumo-shade-40pct: rgba(33, 33, 33, 0.4);
      --lumo-shade-50pct: rgba(33, 33, 33, 0.5);
      --lumo-shade-60pct: rgba(33, 33, 33, 0.6);
      --lumo-shade-70pct: rgba(33, 33, 33, 0.7);
      --lumo-shade-80pct: rgba(33, 33, 33, 0.8);
      --lumo-shade-90pct: rgba(33, 33, 33, 0.9);
      --lumo-primary-color-50pct: rgba(235, 89, 5, 0.5);
      --lumo-primary-color-10pct: rgba(235, 89, 5, 0.1);
      --lumo-error-color-50pct: rgba(231, 24, 24, 0.5);
      --lumo-error-color-10pct: rgba(231, 24, 24, 0.1);
      --lumo-success-color-50pct: rgba(62, 229, 170, 0.5);
      --lumo-success-color-10pct: rgba(62, 229, 170, 0.1);
      --lumo-shade: hsl(0, 0%, 13%);
      --lumo-primary-color: hsl(22, 96%, 47%);
      --lumo-primary-text-color: hsl(22, 100%, 42%);
      --lumo-error-color: hsl(0, 81%, 50%);
      --lumo-error-text-color: hsl(0, 86%, 45%);
      --lumo-success-color: hsl(159, 76%, 57%);
      --lumo-success-contrast-color: hsl(159, 29%, 10%);
      --lumo-success-text-color: hsl(159, 61%, 40%);
      --lumo-font-family: Consolas, "Andale Mono WT", "Andale Mono", "Lucida Console", "Lucida Sans Typewriter", "DejaVu Sans Mono", "Bitstream Vera Sans Mono", "Liberation Mono", "Nimbus Mono L", Monaco, "Courier New", Courier, monospace;
      --lumo-border-radius: calc(var(--lumo-size-m) / 2);
    }
  </style>
</custom-style>
`;

document.head.appendChild($_documentContainer.content);
```

### Build result
target/frontend/generated-flow-imports.js
```
const div = document.createElement('div');
div.innerHTML = '<custom-style><style include="lumo-color lumo-typography"></style></custom-style>';
document.head.insertBefore(div.firstElementChild, document.head.firstChild);
document.body.setAttribute('theme', 'light');

import 'Frontend/styles/shared-styles.js';
import '@vaadin/vaadin-lumo-styles/color.js';
import '@vaadin/vaadin-lumo-styles/typography.js';
import '@vaadin/vaadin-lumo-styles/sizing.js';
import '@vaadin/vaadin-lumo-styles/spacing.js';
import '@vaadin/vaadin-lumo-styles/style.js';
import '@vaadin/vaadin-lumo-styles/icons.js';
import '@vaadin/vaadin-tabs/theme/lumo/vaadin-tab.js';
import '@vaadin/vaadin-tabs/theme/lumo/vaadin-tabs.js';
import '@vaadin/vaadin-app-layout/theme/lumo/vaadin-app-layout.js';
import '@vaadin/vaadin-ordered-layout/theme/lumo/vaadin-vertical-layout.js';
```

**Comment by juhopiirainen** [comment]
Moved to Flow team for investigations.

**Comment by denis-anisimov** [comment]
Confirmed.
To reproduce: use the attached project, run `mvn clean install -DskipTests` , mvn spring-boot:run` and stop the server.

`target/frontend/generated-flow-imports.js` contains on the top which makes it overwritten by Lumo styes.

I guess it happens because JS modules form `Lumo.class` are added after the JS module declared on application class.

It doesn't help if I set `JsModule` below the `Theme` annotation (actually if I recall correctly the code `Theme` annotation is handled in the very end of the processing , that's why all its modules are in the end).

As I understand the solution should be : place all things discovered from `@Theme`  annotation before any discovered `JSModule` annotation. So that `JsModule` annotations from app classes are below anything discovered from `@Theme`.
